### PR TITLE
Add command test cases

### LINKED ISSUE #74: Add Testing
Add test cases for all the new methods to pass CodeCov tests.

### PR DESCRIPTION
**Pull request by shirsho-12**
closes #74

### CODE REVIEW

**Review by jinbesan — COMMENTED** [review]
Looks Good to me

File: src/test/java/seedu/address/commons/stubs/ModelStub.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    @Override
+    public void setExpenseTracker(ReadOnlyExpenseTracker expenseTracker) {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'setExpenseTracker'");
+    }
+
+    @Override
+    public ReadOnlyExpenseTracker getExpenseTracker() {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'getExpenseTracker'");
```
Comment (gitsac) [inline src/test/java/seedu/address/commons/stubs/ModelStub.java:119]: Good method to indicate methods which are not covered by tests yet.

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java
Comment (gitsac) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:1]: Good to edit out traces of AB3 to make it our own product.

**Review by gitsac — APPROVED** [review]
LGTM, please go ahead.

File: src/test/java/seedu/address/logic/commands/AddCategoryCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+public class AddCategoryCommandTest {
+    private UserDefinedCategory toAdd = new UserDefinedCategory("test", "test");
+    private UserDefinedCategory toAdd2 = new UserDefinedCategory("test2", "test2");
+    private ModelStub model = new ModelStub();
+
+    @Test
+    public void addCategoryCommandTest() throws CommandException {
+        requireNonNull(model);
+        model.addCategory(toAdd);
+        model.addCategory(toAdd2);
```
Comment (nicleejy) [inline src/test/java/seedu/address/logic/commands/AddCategoryCommandTest.java:21]: This test case should run an assertion to ensure that the add returns a success message when added.

File: src/test/java/seedu/address/logic/commands/AddExpenseCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    private Expense expense = new Expense("test", 1.0, date, category);
+    private Expense expense2 = new Expense("test2", 2.0, date, category);
+    private Expense expense3 = new Expense("test3", 3.0, date, category);
+
+    @Test
+    public void addExpenseCommandTest() throws CommandException {
+        requireNonNull(model);
+        model.addExpense(expense);
+        model.addExpense(expense2);
+        model.addExpense(expense3);
```
Comment (nicleejy) [inline src/test/java/seedu/address/logic/commands/AddExpenseCommandTest.java:29]: Same goes for this part

**Review by nicleejy — CHANGES_REQUESTED** [review]
Other than small issue in the test case in AddExpense and AddCategory test files, the rest look good!

File: src/test/java/seedu/address/logic/commands/AddCategoryCommandTest.java (hunk 2 above)
Comment (shirsho-12) [reply src/test/java/seedu/address/logic/commands/AddCategoryCommandTest.java:22]: ###

Comment (shirsho-12) [reply src/test/java/seedu/address/logic/commands/AddCategoryCommandTest.java:22]: You are right, fixing it now.

File: src/test/java/seedu/address/logic/commands/AddExpenseCommandTest.java (hunk 3 above)
Comment (shirsho-12) [reply src/test/java/seedu/address/logic/commands/AddExpenseCommandTest.java:30]: You are right, fixing it now.

File: src/test/java/seedu/address/logic/commands/AddCategoryCommandTest.java (hunk 2 above)
Comment (shirsho-12) [reply src/test/java/seedu/address/logic/commands/AddCategoryCommandTest.java:22]: Fixed

**Pull request merged**